### PR TITLE
Fixes petri dishes not changing sprites after being washed

### LINKED ISSUE #65603: Petri dishes dont change sprite after cleaning them out
Pretty much what it says in the title. If you wash out a used petri dish in a sink they will keep the appearance of whatever was in it but will actually be empty when you examine them.

### PR DESCRIPTION
**Pull request by robbertapir**
## About The Pull Request
Fixes #65603.

## Why It's Good For The Game
Bugfix.

## Changelog

:cl:
fix: Petri dish cultures no longer stay behind on the sprite like some sort of ghost after being washed out of it. Cells don't have souls.
/:cl:

### CODE REVIEW

File: code/modules/research/xenobiology/vatgrowing/petri_dish.dm [hunk 1]
```diff
@@ -26,7 +26,8 @@
 	if(!sample || !istype(A, /obj/structure/sink))
 		return FALSE
 	to_chat(user, span_notice("You wash the sample out of [src]."))
-	sample = null
+	QDEL_NULL(sample)
+	update_appearance()
```
Comment (ShizCalev) [inline code/modules/research/xenobiology/vatgrowing/petri_dish.dm:30]: should be handled in handle_atom_del instead.
```
/obj/item/petri_dish/handle_atom_del(atom/deleting_atom)
	if(deleting_atom == sample)
		sample = null
		update_appearance()
	return ..()
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/vatgrowing/petri_dish.dm (hunk 1 above)
Comment (robbertapir) [reply code/modules/research/xenobiology/vatgrowing/petri_dish.dm]: done

File: code/modules/research/xenobiology/vatgrowing/petri_dish.dm [hunk 2]
```diff
@@ -26,7 +37,7 @@
 	if(!sample || !istype(A, /obj/structure/sink))
 		return FALSE
 	to_chat(user, span_notice("You wash the sample out of [src]."))
-	sample = null
+	QDEL_NULL(sample)
```
Comment (Mothblocks) [inline code/modules/research/xenobiology/vatgrowing/petri_dish.dm:40]: You should keep this as `sample = null`--this is a pointer to the original sample, and so you'll be deleting that too.

image

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/vatgrowing/petri_dish.dm (hunk 2 above)
Comment (robbertapir) [reply code/modules/research/xenobiology/vatgrowing/petri_dish.dm:34]: Destroy and the growing vat's plunger_act already QDEL_NULL the sample. Also, are you sure there is an "original sample"? The deposited_sample here is created in the proc try_to_swab from the contents of swabbed_items, which are then cleared, and only stays in the dish. When the sample is deposited into a growing vat, that growing vat creates a copy of the dish's sample. One sample should only be in one place at a time.
Przechwytywanie1
Przechwytywanie

Comment (robbertapir) [reply code/modules/research/xenobiology/vatgrowing/petri_dish.dm]: Anyways the qdel wasn't really necessary to fix the bug so I removed it.

**Pull request merged**